### DISCUSSION TITLE
Clipping image inside a vector

### DISCUSSION THREAD

**Discussion by lucassalazar** [diegomura/react-pdf#1677] [Q&A]
Is there an alternative way to clip an image inside a vector shape using @react-pdf/renderer?
I didn't found the image pdfkit method specified in the docs. 

Using pdfkit:

```
  doc
    .moveTo(0, 0)
    .lineTo(235.28, 0)
    .lineTo(235.28 + 250, 250)
    .lineTo(200, 250)
    .lineTo(0, 50)
    .lineTo(0, 0)
    .save()
    .clip();

  doc
   .image("image_url", 0, 0, { width: 500 })
   .restore();
```
Any ideas?

**Comment by jeetiss** [comment]
Hi @lucassalazar, you can use `svgs` and `clipPath` (see the [example](https://react-pdf.org/repl?code=140630f60760ce02e004b027806c0a6538178e06518b51c00b5551803a1002754043195002806f60e38023300134402e38addbb000eb5bb700961003980804c001800d1b38017d57a809401b982848b0e00113020b1c46dab003e2b6a00f1990015c02daa08316daf68e001568655011f0d1319890d0a1c8b979d4112400bd5130008801040159d37d85851c7000dc64e001dd25b860892201191515124924648861eb1b138b2551ca0084c0003c3314e0c61a271af2fc0ae19d5000cca1f2e60b1c0185912444826ae0aa32dc6001684124a840d066d6d7362eaf424047d3b315d2e0411032de3ea87fde70003d2ad6ef32056c767b22282e68e20498962b031831c00497730550b33588d984d1c5cdbef8f52120a57287d16ae957151908c003131cce0f34369d264e1344d2cc66054aa350124d9470169b46042c68694960f61412e19768c044503e10281924c48562ae6808990b4281112860771024444300c0c0273a800d800cc8a0743be400166c80039edf213ad0402045001d85d41b77716841803f22d241d6a18044003272a612649daba6008e984f7bd39b678462b1b0ff1024a325f2128aa54e655aab56624c65b7317b53a04b04f4fa831798ca6038e583b9913e6d1b632080e150c845809d28b3410c4e5e6e07dd42d82a970a88e5b6fd690dd9520edc23ab94ee74bb5df770e85c14fe9002c9bdc600196c9bbc8aeb767ae00d723c80e87aaf9da8a39061a7ad91fee43da9e8009c41881f238141bc8751fec86287043a706be41981705c11e9c09f9da6860178581647c8f21fed9381f2364885e60057a419fe605d4707640e9e141b906f2fe4a39070681f22be9ea7a30628de9fefc63ac18490eb49419da703c8fc764d9231af91130506d91a9f21da02506a2449fc5d4d91c1e843af4571c05c1525d4418197003a265911e449707906e886e85baca7699ea2184489c67b1fe401d85d43e4896e9d40e9c0da4c176a19ba6f94052529549324d9125491fad1a07818a1d4716897e4febf9da4562876ae19057e6ea28505d5e0479bc4d95fa7a6e9ba70106797596e849604fee960df47c15a4499f83a905c09e899f22119e9e19f928ec77ae41f50ea21064a1ae5c0707217d6ad20499666194b4994eaf1e9489e97a1086f51a7be45539bfbc1e40353fbbef46b56a785f8635ef98c70628afbf502511276f9a0ee1895f9e56d13f5fdeb501bf5017049dfc42def4ba00628fd6dd5fb3aafbba004d1c969dbb5ba17631e07614b5d4f4fedaf959e41da56593ab5399cef9d91255061d3376412773565698370d44673a4759ed54921a3aafb05c2fda700756f183155f95eb6b607e1156be5457a76aa1b96fdf05c5ff9b5a14a5c87da19581c18bafcc91fc6ad0e9db607698d491c17c8a1673bb4d150479dce0176a73676b5b8dd93051112f4dee80dc6699e67dabb69d19e69da44bfaedbb8cb9dcd3a884d17e69dea4116260be0625c97fe5c4f194d65ae6fe5cea14c7b7bf5b550691e46f1fb75366461bf761b84e1ac6f36cec108743f465bda47100501a9ff6001687c20a7265a1ed08def0aeec8a72e896ad8aca701e29dacac49df60852478d4192d2f4932e78b257aa0ecbef6101051cfc81b24a150a29502b4768a03a53ff794201150c0654aa9d526a2c43a8f501a234268cd05a2b4369ed23448a751208e10742b8aca7a5a04e5f083a6e0a80631c64c0099932a674c9991a1261cc798872dc42c4098b0846dcf082b1563b8159806bf26c5d1c06408e85229a1c06ec0318628c7181c50731f53e379fc21f63c871d23324bc8f06e35f7987781f23e10e2255781150ad85b206c32e40d9b8c50b6e31aba112c2275e98c977150473aa91b21b1fda342c2e8500b73042034ea3210f4b257dafd0686a4b8aed3cc0341279506a1b0ac4c490cee4d9a64a32525b8ad1186cb4ad8a4c9224cbdbe93b47691c5813cc8ecdd2f909aee5999d8ed6ca480b1914ad8cbd32520c6cdbd16495aa851d8b10d28b4224b91b2274d9bf560cea58a6af2aafd43d06c2c6a845a4710593f852af936a40438b74bc9d657e8f101a6b3fa838f39472068fd61664506b29776c94769055744b54897125a604a6438aaaf69f65991828057f13972ef0448ab4e5aea200a49772ad2868388686cdb09096452f594a3525955d4eaa96f952d7195762a4183621d0fc619d4b3b262b544cbf92627f90e4a28391738e7fe4225a43630b72e313d48891896e8d49055ea0cddc6817bacb3ac6d15b1664ec8ef604da3c12e89a8c7c11122511b702f9627feb7d372dc07e06ad633f6846fce9232431ac97f8f0e1ce10791f27ac829d448a36c129d44c0ebe70210520b541a92fba0a80fa90d31a7003832d35a5b40a43d87a0329e97d2a0474751162285a068b68430f8c541130a634c1c3d86284e1b99f31823e102350108f2ca51613c22088229c0227301e0bc0f86007a00c000253a020060004130000c5c80d0080742a8230670e60957b683033a800000)) or access `pdfkit` internals with `Canvas` component

**Reply by asingleton6** [reply]
@jeetiss I tried replicating the same example on my end, but I ran into this TypeScript error. Is this the expected behavior?

![Screenshot 2024-03-25 at 4 12 58 PM](https://github.com/diegomura/react-pdf/assets/5387554/2a7fde75-461e-4ec5-a99c-8fd58bdab754)
